File: TCS NQT Prep/Java/q1.java
// import java.util.*;
// class q1{

//     public static void main(String [] args){
//         System.out.print("Enter the number : ");
//         Scanner sc = new Scanner(System.in);
//         int n=sc.nextInt();
//         int sum = Natural_Prod_Sum(n);
//         System.out.println("the  sum of product of 1st 10 natural numbers with the given number is "+sum); 
//         sc.close();
//     }
//     public static int Natural_Prod_Sum(int m){
//         int sum=0;
//         for(int i=1;i<=10;i++){
//             sum+=m*i;
//         }
//         return sum;
//     }
// }

//time complexity of this code is O(1) as it is not dependent on the input size. The space complexity is O(1) as we are using only a few variables.



import java.util.*;
class q1{

    public static void main(String [] args){
        System.out.print("Enter the number : ");
        Scanner sc = new Scanner(System.in);
        int n=sc.nextInt();
        int sum = n*55;
        System.out.println("the  sum of product of 1st 10 natural numbers with the given number is "+sum); 
        sc.close();
    }
}

// The time complexity of this code is O(1) as it is not dependent on the input size. The space complexity is O(1) as we are using only a few variables.

File: TCS NQT Prep/cpp/q1.cpp
// #include<bits/stdc++.h>
// using namespace std;

// int Natural_Num_Sum(int a){

//     if(a==0){
//         return 0;
//     }
    
//     int sum =0;


//         for(int i=1;i<=10;i++){
//             sum+= i*a;
//         }
//         return sum;
// }

// int main(){
//     int n;
//     cout<<"Enter The Number : ";
//     cin>>n;

//     int n1 = Natural_Num_Sum(n);
//     cout <<"The Sum Of First "<< n<<"Natural Numbers is : "<<n1<<endl;
//     return 0;
// }

//    this code is of Time Complexity: O(1) but it is not efficient...


// // 2nd Approach
#include<bits/stdc++.h>
using namespace std;

int main(){

    int n;
    cout<<"number de....";
    cin>>n;

    int sum = n*55;

    cout<<"The Sum Of First "<<n<<" Natural Numbers is : "<<sum<<endl;
    return 0;

}

// Time Complexity: O(1) so it is efficient....



//Key Takeaways:
//try to simplify the logic mathematically to reduce the time complexity of the code.
//always use #include <bits/stdc++.h> in your code to avoid writing multiple header files.
//as #include <bits/stdc++.h> basiccally includes all the header files.
//always use namespace std; in your code to avoid writing std:: in front of every cin, cout, etc.